crates/core/src/diff/fragment/tests.rs

```diff
@@ -0,0 +1,728 @@
+                    "d": [
+                        [
+                            1
+                        ],
+                        [
+                            2
+                        ],
+                        [
+                            3
+                        ]
+                    ],
+                    "s": [
+                        "\n  ",
+                        "\n"
+                    ]
+                },
+                "s": [
+                    "",
+                    ""
+                ]
+            },
+            "c": {
+                "1": {
+                    "0": {
+                        "d": [
+                            [
+                                "3"
+                            ],
+                            [
+                                "4"
+                            ],
+                            [
+                                "5"
+                            ]
+                        ],
+                        "s": [
+                            "\n    <Text>Item ",
+                            "</Text>\n"
+                        ]
+                    },
+                    "s": [
+                        "<Group>\n",
+                        "\n</Group>"
+                    ]
+                },
+                "2": {
+                    "0": {
+                        "d": [
+                            [
+                                "6"
+                            ],
+                            [
+                                "7"
+                            ],
+                            [
+                                "8"
+                            ]
+                        ]
+                    },
+                    "s": 1
+                },
+                "3": {
+                    "0": {
+                        "d": [
+                            [
+                                "9"
+                            ],
+                            [
+                                "10"
+                            ],
+                            [
+                                "11"
+                            ]
+                        ]
+                    },
+                    "s": 1
+                }
+            },
+            "s": [
+                "<div>",
+                "</div>"
+            ]
+        }"#;
+    let root: RootDiff = serde_json::from_str(input_json).expect("Failed to deserialize fragment");
+    //println!("{root:#?}");
+    let root: Root = root.try_into().expect("Failed to convert RootDiff to Root");
+    println!("root diff: {root:#?}");
+    let out: String = root.try_into().expect("Failed to convert Root into string");
+    println!("out: {out}");
+    let expected = r#"<div>
+  <Group>
+
+    <Text>Item 3</Text>
+
+    <Text>Item 4</Text>
+
+    <Text>Item 5</Text>
+
+</Group>
+
+  <Group>
+
+    <Text>Item 6</Text>
+
+    <Text>Item 7</Text>
+
+    <Text>Item 8</Text>
+
+</Group>
```

Comment:
Why isn't this ending tag indented the same as the starting tag?

Reply:
This was just templating issue. I've updated the template to be correct and then the unit tests as well.